### ISSUE TITLE
[telegraf-operator] Unable to update sidecar secrets when classes changes

### ISSUE THREAD

**Issue by JoshuaWatt** [influxdata/helm-charts#455]
The telegraf operator doesn't update the sidecar secrets when the base classes secret changes. It looks like it is trying to enumerate the secrets, but is unable to because it is missing the permission to list namespaces

In the telegraf-operator pod log, I see the following:
```
2022-03-28T15:20:23.164Z	ERROR	updater	unable to list namespaces	{"error": "namespaces is forbidden: User \"system:serviceaccount:telegraf-operator:telegraf-operator\" cannot list resource \"namespaces\" in API group \"\" at the cluster scope"}
github.com/go-logr/zapr.(*zapLogger).Error
	/go/pkg/mod/github.com/go-logr/zapr@v0.2.0/zapr.go:132
main.(*secretsUpdater).onChange
	/workspace/updater.go:52
main.(*telegrafClassesWatcher).batchChanges
	/workspace/watcher.go:110
```

My values.yaml looks like:
```
# Clear out the default configuration; we will manually provide a
# secret since it has the database password
classes:
  data: ""
```

**Comment by wojciechka** [comment]
Oh interesting. Yes, this is a permission issue so can be fixed in this repo without having to update telegraf-operator.

**Comment by alespour** [comment]
Was this fixed in #454?
